### PR TITLE
Avoid NullReferenceException in MissingMetadataTypeSymbol if it doesn't have containing module.

### LINKED ISSUE #52516: Structs referenced from a .winmd file crash the compiler if they have a field member which uses a nested type
**Version Used**: 

dotnet: 5.0.201
Microsoft.CodeAnalysis: 3.8.0
Microsoft.CodeAnalysis.CSharp: 3.8.0

**Steps to Reproduce**:

My repro is a little different (see below) since it is building generated C# with Roslyn, but I think these simplified steps should demonstrate the bug:

1. Build the `Windows.Win32.winmd` with https://github.com/microsoft/win32metadata, or download it as a NuGet package from https://www.nuget.org/packages/Microsoft.Windows.SDK.Win32Metadata/.
2. Add a reference to Windows.Win32.winmd to your project.
3. Write a C# program that references `Windows.Win32.Automation.VARIANT` in a method parameter.
4. Attempt to build the solution with the `dotnet` command.

**Expected Behavior**:

It should build.

The `Windows.Win32.winmd` fully defines the `Windows.Win32.Automation.VARIANT` type, it uses nested types for the structs and unions that show up as `_Anonymous_e__Union` and `_Anonymous_e__Struct`. The nested types are fully defined in the VS Object Browser after adding a reference to the file. Here's an example of the C# it generated with `ClangSharpPInvokeGenerator` (with paths redacted):
```cs
    public unsafe partial struct VARIANT
    {
        [NativeTypeName("tagVARIANT::(anonymous union at .../win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/oaidl.h:470:5)")]
        public _Anonymous_e__Union Anonymous;

        [StructLayout(LayoutKind.Explicit)]
        public unsafe partial struct _Anonymous_e__Union
        {
            [FieldOffset(0)]
            [NativeTypeName("tagVARIANT::(anonymous struct at .../win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/oaidl.h:472:9)")]
            public _Anonymous_e__Struct Anonymous;

            [FieldOffset(0)]
            public DECIMAL decVal;

            public unsafe partial struct _Anonymous_e__Struct
            {
                [NativeTypeName("VARTYPE")]
                public ushort vt;

                [NativeTypeName("WORD")]
                public ushort wReserved1;

                [NativeTypeName("WORD")]
                public ushort wReserved2;

                [NativeTypeName("WORD")]
                public ushort wReserved3;

                [NativeTypeName("tagVARIANT::(anonymous union at .../win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/oaidl.h:478:13)")]
                public _Anonymous_e__Union Anonymous;

                [StructLayout(LayoutKind.Explicit)]
                public unsafe partial struct _Anonymous_e__Union
                {
                    [FieldOffset(0)]
                    [NativeTypeName("LONGLONG")]
                    public long llVal;

                    [FieldOffset(0)]
                    [NativeTypeName("LONG")]
                    public int lVal;

                    [FieldOffset(0)]
                    [NativeTypeName("BYTE")]
                    public byte bVal;

                    [FieldOffset(0)]
                    [NativeTypeName("SHORT")]
                    public short iVal;

                    [FieldOffset(0)]
                    [NativeTypeName("FLOAT")]
                    public float fltVal;

                    [FieldOffset(0)]
                    [NativeTypeName("DOUBLE")]
                    public double dblVal;

                    [FieldOffset(0)]
                    [NativeTypeName("VARIANT_BOOL")]
                    public short boolVal;

                    [FieldOffset(0)]
                    [NativeTypeName("VARIANT_BOOL")]
                    public short __OBSOLETE__VARIANT_BOOL;

                    [FieldOffset(0)]
                    [NativeTypeName("SCODE")]
                    public int scode;

                    [FieldOffset(0)]
                    public CY cyVal;

                    [FieldOffset(0)]
                    [NativeTypeName("DATE")]
                    public double date;

                    [FieldOffset(0)]
                    [NativeTypeName("BSTR")]
                    public ushort* bstrVal;

                    [FieldOffset(0)]
                    [NativeTypeName("IUnknown *")]
                    public IUnknown* punkVal;

                    [FieldOffset(0)]
                    [NativeTypeName("IDispatch *")]
                    public IDispatch* pdispVal;

                    [FieldOffset(0)]
                    [NativeTypeName("SAFEARRAY *")]
                    public SAFEARRAY* parray;

                    [FieldOffset(0)]
                    [NativeTypeName("BYTE *")]
                    public byte* pbVal;

                    [FieldOffset(0)]
                    [NativeTypeName("SHORT *")]
                    public short* piVal;

                    [FieldOffset(0)]
                    [NativeTypeName("LONG *")]
                    public int* plVal;

                    [FieldOffset(0)]
                    [NativeTypeName("LONGLONG *")]
                    public long* pllVal;

                    [FieldOffset(0)]
                    [NativeTypeName("FLOAT *")]
                    public float* pfltVal;

                    [FieldOffset(0)]
                    [NativeTypeName("DOUBLE *")]
                    public double* pdblVal;

                    [FieldOffset(0)]
                    [NativeTypeName("VARIANT_BOOL *")]
                    public short* pboolVal;

                    [FieldOffset(0)]
                    [NativeTypeName("VARIANT_BOOL *")]
                    public short* __OBSOLETE__VARIANT_PBOOL;

                    [FieldOffset(0)]
                    [NativeTypeName("SCODE *")]
                    public int* pscode;

                    [FieldOffset(0)]
                    [NativeTypeName("CY *")]
                    public CY* pcyVal;

                    [FieldOffset(0)]
                    [NativeTypeName("DATE *")]
                    public double* pdate;

                    [FieldOffset(0)]
                    [NativeTypeName("BSTR *")]
                    public ushort** pbstrVal;

                    [FieldOffset(0)]
                    [NativeTypeName("IUnknown **")]
                    public IUnknown** ppunkVal;

                    [FieldOffset(0)]
                    [NativeTypeName("IDispatch **")]
                    public IDispatch** ppdispVal;

                    [FieldOffset(0)]
                    [NativeTypeName("SAFEARRAY **")]
                    public SAFEARRAY** pparray;

                    [FieldOffset(0)]
                    [NativeTypeName("VARIANT *")]
                    public VARIANT* pvarVal;

                    [FieldOffset(0)]
                    [NativeTypeName("PVOID")]
                    public void* byref;

                    [FieldOffset(0)]
                    [NativeTypeName("CHAR")]
                    public sbyte cVal;

                    [FieldOffset(0)]
                    [NativeTypeName("USHORT")]
                    public ushort uiVal;

                    [FieldOffset(0)]
                    [NativeTypeName("ULONG")]
                    public uint ulVal;

                    [FieldOffset(0)]
                    [NativeTypeName("ULONGLONG")]
                    public ulong ullVal;

                    [FieldOffset(0)]
                    [NativeTypeName("INT")]
                    public int intVal;

                    [FieldOffset(0)]
                    [NativeTypeName("UINT")]
                    public uint uintVal;

                    [FieldOffset(0)]
                    [NativeTypeName("DECIMAL *")]
                    public DECIMAL* pdecVal;

                    [FieldOffset(0)]
                    [NativeTypeName("CHAR *")]
                    public sbyte* pcVal;

                    [FieldOffset(0)]
                    [NativeTypeName("USHORT *")]
                    public ushort* puiVal;

                    [FieldOffset(0)]
                    [NativeTypeName("ULONG *")]
                    public uint* pulVal;

                    [FieldOffset(0)]
                    [NativeTypeName("ULONGLONG *")]
                    public ulong* pullVal;

                    [FieldOffset(0)]
                    [NativeTypeName("INT *")]
                    public int* pintVal;

                    [FieldOffset(0)]
                    [NativeTypeName("UINT *")]
                    public uint* puintVal;

                    [FieldOffset(0)]
                    [NativeTypeName("tagVARIANT::(anonymous struct at .../win32metadata/artifacts/InstalledPackages/Microsoft.Windows.SDK.CPP.10.0.19041.5/c/Include/10.0.19041.0/um/oaidl.h:525:17)")]
                    public _Anonymous_e__Struct Anonymous;

                    public unsafe partial struct _Anonymous_e__Struct
                    {
                        [NativeTypeName("PVOID")]
                        public void* pvRecord;

                        [NativeTypeName("IRecordInfo *")]
                        public IRecordInfo* pRecInfo;
                    }
                }
            }
        }
    }
```

**Actual Behavior**:

I found this while looking into https://github.com/microsoft/win32metadata/issues/66. My fork/branch at https://github.com/wravery/win32metadata/tree/external-pkgs demonstrates the problem, it can't compile the C# for the second `Microsoft.Web.WebView2.winmd` with a reference to the first one. The same generated C# code works fine if it's built directly into first `Windows.Win32.winmd`, but then it's just adding to the monolith.

Compilation fails with a `System.NullReferenceException` because it assigns an `ErrorType` to the `Anonymous` field, and then it can't get the containing assembly to return the `ErrorInfo`. I traced it back to here, where I think the bug lives:
```
>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PENamedTypeSymbol.LoadMembers() Line 1316	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PENamedTypeSymbol.EnsureAllMembersAreLoaded() Line 1158	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PENamedTypeSymbol.GetMembers() Line 790	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamespaceOrTypeSymbol.GetMembersUnordered() Line 119	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.GetInstanceFieldsAndEvents() Line 549	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.BaseTypeAnalysis.DependsOnDefinitelyManagedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol type, System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbol> partialClosure, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 163	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.BaseTypeAnalysis.GetManagedKind(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol type, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 129	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol.GetManagedKind(ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 407	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.CheckManagedAddr(Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol type, Microsoft.CodeAnalysis.Location location, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1321	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindNamespaceOrTypeOrAliasSymbol.__bindPointer|940_4(ref Microsoft.CodeAnalysis.CSharp.Binder.<>c__DisplayClass940_0 value) Line 576	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindNamespaceOrTypeOrAliasSymbol(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Roslyn.Utilities.ConsList<Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol> basesBeingResolved, bool suppressUseSiteDiagnostics) Line 436	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Binder.BindTypeOrAlias(Microsoft.CodeAnalysis.CSharp.Syntax.ExpressionSyntax syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, Roslyn.Utilities.ConsList<Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol> basesBeingResolved, bool suppressUseSiteDiagnostics) Line 318	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ParameterHelpers.MakeParameters<Microsoft.CodeAnalysis.CSharp.Syntax.ParameterSyntax, Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol, Microsoft.CodeAnalysis.CSharp.Symbol>(Microsoft.CodeAnalysis.CSharp.Binder binder, Microsoft.CodeAnalysis.CSharp.Symbol owner, Microsoft.CodeAnalysis.SeparatedSyntaxList<Microsoft.CodeAnalysis.CSharp.Syntax.ParameterSyntax> parametersList, out Microsoft.CodeAnalysis.SyntaxToken arglistToken, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, bool allowRefOrOut, bool allowThis, bool addRefReadOnlyModifier, bool suppressUseSiteDiagnostics, int lastIndex, System.Func<Microsoft.CodeAnalysis.CSharp.Binder, Microsoft.CodeAnalysis.CSharp.Symbol, Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations, Microsoft.CodeAnalysis.CSharp.Syntax.ParameterSyntax, Microsoft.CodeAnalysis.RefKind, int, Microsoft.CodeAnalysis.SyntaxToken, Microsoft.CodeAnalysis.SyntaxToken, bool, Microsoft.CodeAnalysis.DiagnosticBag, Microsoft.CodeAnalysis.CSharp.Symbols.ParameterSymbol> parameterCreationFunc, bool parsingFunctionPointer) Line 177	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ParameterHelpers.MakeParameters(Microsoft.CodeAnalysis.CSharp.Binder binder, Microsoft.CodeAnalysis.CSharp.Symbol owner, Microsoft.CodeAnalysis.CSharp.Syntax.BaseParameterListSyntax syntax, out Microsoft.CodeAnalysis.SyntaxToken arglistToken, Microsoft.CodeAnalysis.DiagnosticBag diagnostics, bool allowRefOrOut, bool allowThis, bool addRefReadOnlyModifier) Line 27	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceDelegateMethodSymbol.InvokeMethod.InvokeMethod(Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol delegateType, Microsoft.CodeAnalysis.RefKind refKind, Microsoft.CodeAnalysis.CSharp.Symbols.TypeWithAnnotations returnType, Microsoft.CodeAnalysis.CSharp.Syntax.DelegateDeclarationSyntax syntax, Microsoft.CodeAnalysis.CSharp.Binder binder, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 253	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceDelegateMethodSymbol.AddDelegateMembers(Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol delegateType, Microsoft.CodeAnalysis.PooledObjects.ArrayBuilder<Microsoft.CodeAnalysis.CSharp.Symbol> symbols, Microsoft.CodeAnalysis.CSharp.Syntax.DelegateDeclarationSyntax syntax, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 64	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.AddDeclaredNontypeMembers(Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.MembersAndInitializersBuilder builder, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 2538	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.BuildMembersAndInitializers(Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 2475	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetMembersAndInitializers() Line 1349	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.MakeAllMembers(Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 2281	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetMembersByNameSlow() Line 1379	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetMembersByName() Line 1371	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetMembersUnordered() Line 1213	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ComputeNullableContextValue() Line 3978	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetLocalNullableContextValue() Line 3950	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.GetCommonNullableValues(Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation, ref Microsoft.CodeAnalysis.CSharp.MostCommonNullableValueBuilder builder) Line 1396	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ComputeNullableContextValue() Line 3978	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetLocalNullableContextValue() Line 3950	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.GetNullableContextValue() Line 1381	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.ShouldEmitNullableContextValue(out byte value) Line 1456	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberMethodSymbol.AfterAddingTypeMembersChecks(Microsoft.CodeAnalysis.CSharp.ConversionsBase conversions, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 819	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceOrdinaryMethodSymbolBase.AfterAddingTypeMembersChecks(Microsoft.CodeAnalysis.CSharp.ConversionsBase conversions, Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 630	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.CheckSpecialMemberErrors(Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1841	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.AfterMembersChecks(Microsoft.CodeAnalysis.DiagnosticBag diagnostics) Line 1414	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ForceComplete(Microsoft.CodeAnalysis.SourceLocation locationOpt, System.Threading.CancellationToken cancellationToken) Line 539	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbol.ForceCompleteMemberByLocation(Microsoft.CodeAnalysis.SourceLocation locationOpt, Microsoft.CodeAnalysis.CSharp.Symbol member, System.Threading.CancellationToken cancellationToken) Line 802	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.ForceComplete.AnonymousMethod__0(int i) Line 61	C#
 	Microsoft.CodeAnalysis.dll!Roslyn.Utilities.UICultureUtilities.WithCurrentUICulture.AnonymousMethod__0(int param) Line 173	C#
```

This does not repro if you compile these files together, as in the default build for https://github.com/microsoft/win32metadata. It references an entirely different `TypeSymbol` in that case which does see the nested type for that field:
```
>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetMembersAndInitializers() Line 1342	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetInstanceFieldsAndEvents() Line 1399	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.BaseTypeAnalysis.DependsOnDefinitelyManagedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol type, System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbol> partialClosure, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 163	C#
```

For anything other an an `enum`, the [PENamedTypeSymbol.LoadMembers()](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1168) method loads in this order:

1. [fields](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1227)
2. [methods](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1240) (potentially including a [synthetic parameterless constructor](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1258))
3. [properties](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1262)
4. [events](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1263)

After that, whether it's an `enum` or not, it [loads](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1292) the nested types and [sorts](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs#L1298) them by row ID at the end of the `members` `ArrayBuilder`.

**What seems to be missing is a fixup pass to apply those nested types to any of the struct members which might be referencing them.**

For the sake of completeness, [here](https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs#L73) is where it ends up throwing the `System.NullReferenceException`, because `containingAssembly` returns `null`:
```
>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.MissingMetadataTypeSymbol.ErrorInfo.get() Line 75	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.ErrorTypeSymbol.GetUseSiteDiagnostic() Line 85	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbolExtensions.AddUseSiteDiagnostics(Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol type, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 1383	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.Symbols.BaseTypeAnalysis.DependsOnDefinitelyManagedType(Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol type, System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.CSharp.Symbol> partialClosure, ref System.Collections.Generic.HashSet<Microsoft.CodeAnalysis.DiagnosticInfo> useSiteDiagnostics) Line 194	C#
```

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #52516.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review a small bug fix.

**Review by 333fred — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ internal override DiagnosticInfo ErrorInfo
                 // Since we do not know what task was being performed, for now we just report a generic
                 // "you must add a reference" error.
 
-                if (containingAssembly.IsMissing)
+                if (containingAssembly?.IsMissing == true)
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs:73]: Did you consider marking `this.ContainingAssembly` as having a `AssemblySymbol?` return?

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs:73]: No. It is defined in a file that is not annotated. The problem wasn't that we didn't know that the ContainingAssembly can return null in general. The problem was that we assumed that that cannot happen for this specific flavor of symbol.

File: src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs [hunk 2]
```diff
@@ -94,25 +94,37 @@ internal override DiagnosticInfo ErrorInfo
...
+                        {
+                            // The more straightforward scenario is that we compiled Beta against a version of Alpha
+                            // that had C, and then added a reference to a different version of Alpha that
+                            // lacks the type C:
+
+                            // error CS7069: Reference to type 'C' claims it is defined in 'Alpha', but it could not be found
+                            return new CSDiagnosticInfo(ErrorCode.ERR_MissingTypeInAssembly, this, containingAssembly.Name);
+                        }
+                    }
+                    else if (ContainingType is ErrorTypeSymbol { ErrorInfo: { } info})
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/MissingMetadataTypeSymbol.cs:120]:
```suggestion
                    else if (ContainingType is ErrorTypeSymbol { ErrorInfo: { } info })
```

**Pull request merged**